controllers/upgrade.go

```diff
@@ -326,12 +324,16 @@ func (r *RollingUpgradeContext) SelectTargets(scalingGroup *autoscaling.Group) [
 
 	var unavailableInt int
 	if batchSize.Type == intstr.String {
-		unavailableInt, _ = intstr.GetValueFromIntOrPercent(&batchSize, totalNodes, true)
+		if strings.Contains(batchSize.StrVal, "%") {
```

Comment:
To make sure we call `GetValueFromIntOrPercent` only when the value is a percentage.